host_core/lib/host_core/providers/provider_module.ex

```diff
@@ -75,12 +79,87 @@ defmodule HostCore.Providers.ProviderModule do
     {:noreply, state}
   end
 
+  @impl true
+  def handle_info(:do_health, state) do
+    topic = "wasmbus.rpc.#{state.public_key}.#{state.link_name}.health"
+    payload = %{placeholder: true} |> Msgpax.pack!() |> IO.iodata_to_binary()
+
+    res =
+      try do
+        Gnat.request(:lattice_nats, topic, payload, receive_timeout: 2_000)
+      rescue
+        _e -> {:error, "Received no response on health check topic from provider"}
+      end
+
+    case res do
+      {:ok, _body} -> publish_health_passed(state)
+      {:error, _} -> publish_health_failed(state)
+    end
+
+    Process.send_after(self(), :do_health, @thirty_seconds)
+    {:noreply, state}
+  end
+
   def handle_info({_ref, msg}, state) do
     Logger.info(msg)
 
     {:noreply, state}
   end
 
+  defp publish_health_passed(state) do
```

Comment:
Feels like we _could_ refactor the publish_health_passed|failed here since it's mostly the same between actors and providers, but I'm less of a fan of creating a file to share between the two modules just for this purpose. Just a callout if we refactor it later.